packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1623,12 +1631,14 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   bool get isEndDrawerOpen => _endDrawerOpened;
 
   void _drawerOpenedCallback(bool isOpened) {
+    widget.drawerCallback?.call(isOpened);
```

Comment:
This should be called after setState, otherwise they may get conflicted result when they call Scafford.currentState. isDrawerOpen in the callback